src/models/applicationCost.model.js

```diff
@@ -80,6 +80,9 @@ module.exports = class ApplicationCost {
       .concat(wasteAssessmentApplicationCostItems)
       .sort((a, b) => a.displayOrder - b.displayOrder)
 
+    // Remove cached application data to prevent incorrect total issue
+    delete entityContextToUse.application
+
```

Comment:
This looks like it should all work OK as the subsequent call to `Application.getById` reinstates the `application` object in the context.
However, as this has been highlighted by an observed bug, there should be a test added to verify that the bug is fixed and that this behaviour is functioning correctly.

Reply:
I've added a test but I'm not sure it's the best way of doing it. The problem is we need to clear the cached application data to prevent the behaviour of a function buried a few levels deep -- I started writing a "better" test but it requires a lot of additional mocking and stubbing, and I don't know if that's a good way of doing it -- the test requires deep knowledge of how the function works which I'm not sure is the best idea?